server/bin/unittests

```diff
@@ -259,12 +259,13 @@ function _verify_output {
 
 function _create_user {
     name=${1}
-    # NOTE: we ignore failures here deliberately; it'll always fail in
-    # the 0[.x] tests because of intentionally corrupted configuration,
-    # and unintentional failure in other tests will result in visible
-    # symptoms later.
+    # NOTE: we ignore chatty stdout messages, but capture error output
+    # and fail if the command fails.
     pbench-user-create --username=${name} --email=${name}@example.com \
-        --first-name=${name^} --last-name=User --password=password >/dev/null 2>&1
+        --first-name=${name^} --last-name=User --password=password >/dev/null
+    if [[ ${?} -ne 0 ]]; then
+        exit 101
+    fi
 }
```

Comment:
Contrary to the `NOTE`, you're not actually capturing the error output (here) -- presumably it escapes and is captured by the invoking context.  If that's your intention, it would probably be good to correct/clarify the comment; otherwise...its not hard to actually capture the error output, and then, in the event of an error, do something useful with it.

Reply:
The problem with writing a comment first is that it's easy to miss updating it as the code evolves.

The worst thing is that this appears to have completely broken the unit tests. I'd mentioned I was having troubles with the tox environment on my laptop; I decided to just push this PR and see what happened, but the `unittests` are failing the same way in Jenkins that they are for me, so there's something more systemic apparently triggered by this change... though I can't immediately see what it is. This, of course, was the original motivation for the branch that became this PR, so it'd be ironic if I ended up just pulling it to get the rest through...

Reply:
Are you working with Nikhil's changes to move the database out of memory to a temporary file for the unit tests which need it to persist between invocations?

Reply:
Not relevant for the gold tests; we use an sqlite3 DB.